x/marker/keeper/msg_server.go

```diff
@@ -694,3 +694,39 @@ func (k msgServer) UpdateRequiredAttributes(goCtx context.Context, msg *types.Ms
 
 	return &types.MsgUpdateRequiredAttributesResponse{}, nil
 }
+
+func (k msgServer) UpdateForcedTransfer(goCtx context.Context, msg *types.MsgUpdateForcedTransferRequest) (*types.MsgUpdateForcedTransferResponse, error) {
+	if msg.Authority != k.GetAuthority() {
+		return nil, errors.Wrapf(govtypes.ErrInvalidSigner, "expected %s got %s", k.GetAuthority(), msg.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	marker, err := k.GetMarkerByDenom(ctx, msg.Denom)
+	if err != nil {
+		return nil, fmt.Errorf("could not get marker for %s: %w", msg.Denom, err)
+	}
+
+	if marker.GetMarkerType() != types.MarkerType_RestrictedCoin {
+		return nil, fmt.Errorf("cannot update forced transfer on unrestricted marker %s", msg.Denom)
+	}
+
+	if !marker.HasGovernanceEnabled() {
+		return nil, fmt.Errorf("%s marker does not allow governance control", msg.Denom)
+	}
+
+	if marker.AllowsForcedTransfer() == msg.AllowForcedTransfer {
```

Comment:
I'm on the  fence about this of the force transfer flag.  Do we really care at this point?  I'm just wondering about the case that the state changes between when the gov proposal is submitted and when it runs.  The delayed setting of this state make this a bit problematic.  If this was just a normal tx with immediate feedback this would be fine imo.

Reply:
Since the flag can only change via governance proposal ... this failure would indicate an invalid proposal (setting something already set, user probably made a mistake) ... or perhaps as you mention, two different overlapping governance proposals (one of these shouldn't have happened) ... also feels like a user mistake/error...

Reply:
I waffled on this too.

One one hand, it's similar behavior as most of our stuff.

One the other hand, the reason for a gov prop failure can be a bit of a pain to track down. If a prop fails for this reason, it's kind of a false flag. I.e. it shows as a failure even though state is as the proposal requested.

In the end, I went with our standard behavior. I was really on the fence about it though, and it'd be easy to take out if we decide we don't want that check in this endpoint.